From 8ae51b93efeb8e4fa4210b803e03a4ac17227156 Mon Sep 17 00:00:00 2001
From: Ricky CX Wu <ricky.cx.wu.wiwynn@gmail.com>
Date: Mon, 5 Aug 2024 10:01:07 +0800
Subject: [PATCH 04/22] configurations: yosemite4: Add Sentinel Dome with
 Retimer configs

Add configurations of Sentinel Dome with Retimer for both T1 and T2
configurations.

Tested:
Checked that the configs could be probed according to the FRU
information.

Change-Id: I5ed0067b77a8713eb73c89beb71f68d2b29d1bf7
Signed-off-by: Ricky CX Wu <ricky.cx.wu.wiwynn@gmail.com>
---
 configurations/yosemite4_sentineldome_t1.json |   2 +-
 .../yosemite4_sentineldome_t1_retimer.json    | 372 +++++++++++++++
 configurations/yosemite4_sentineldome_t2.json |  41 +-
 .../yosemite4_sentineldome_t2_retimer.json    | 431 ++++++++++++++++++
 meson.build                                   |   2 +
 5 files changed, 807 insertions(+), 41 deletions(-)
 create mode 100644 configurations/yosemite4_sentineldome_t1_retimer.json
 create mode 100644 configurations/yosemite4_sentineldome_t2_retimer.json

diff --git a/configurations/yosemite4_sentineldome_t1.json b/configurations/yosemite4_sentineldome_t1.json
index 38ba778..69063de 100644
--- a/configurations/yosemite4_sentineldome_t1.json
+++ b/configurations/yosemite4_sentineldome_t1.json
@@ -331,7 +331,7 @@
         }
     ],
     "Name": "Yosemite 4 Sentinel Dome T1 Slot $bus % 15",
-    "Probe": "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': 'Sentinel Dome', 'PRODUCT_PRODUCT_NAME': 'Yosemite V4', 'PRODUCT_INFO_AM2': 'Yosemite V4 T1'})",
+    "Probe": "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': 'Sentinel Dome without Retimer', 'PRODUCT_PRODUCT_NAME': 'Yosemite V4', 'PRODUCT_INFO_AM2': 'Yosemite V4 T1'})",
     "Type": "Board",
     "xyz.openbmc_project.Inventory.Decorator.Asset": {
         "Manufacturer": "$PRODUCT_MANUFACTURER",
diff --git a/configurations/yosemite4_sentineldome_t1_retimer.json b/configurations/yosemite4_sentineldome_t1_retimer.json
new file mode 100644
index 0000000..a3ce02e
--- /dev/null
+++ b/configurations/yosemite4_sentineldome_t1_retimer.json
@@ -0,0 +1,372 @@
+{
+    "Exposes": [
+        {
+            "CheckHysteresisWithSetpoint": true,
+            "Class": "temp",
+            "DCoefficient": 0.0,
+            "FFGainCoefficient": 0.0,
+            "FFOffCoefficient": 0.0,
+            "ICoefficient": -0.035,
+            "ILimitMax": 100,
+            "ILimitMin": -50,
+            "Inputs": [
+                "MB_DIMM_A_TEMP_C_5 $bus % 15 * 10",
+                "MB_DIMM_B_TEMP_C_6 $bus % 15 * 10",
+                "MB_DIMM_C_TEMP_C_7 $bus % 15 * 10",
+                "MB_DIMM_D_TEMP_C_8 $bus % 15 * 10",
+                "MB_DIMM_E_TEMP_C_9 $bus % 15 * 10",
+                "MB_DIMM_G_TEMP_C_11 $bus % 15 * 10",
+                "MB_DIMM_H_TEMP_C_12 $bus % 15 * 10",
+                "MB_DIMM_I_TEMP_C_13 $bus % 15 * 10",
+                "MB_DIMM_J_TEMP_C_14 $bus % 15 * 10",
+                "MB_DIMM_K_TEMP_C_15 $bus % 15 * 10"
+            ],
+            "Name": "PID_MB_DIMM $bus % 15 * 10",
+            "NegativeHysteresis": 2.0,
+            "OutLimitMax": 100,
+            "OutLimitMin": 0,
+            "PCoefficient": -3.0,
+            "PositiveHysteresis": 0.0,
+            "SetPoint": 75.0,
+            "SlewNeg": 0.0,
+            "SlewPos": 0.0,
+            "Type": "Pid",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "CheckHysteresisWithSetpoint": true,
+            "Class": "temp",
+            "DCoefficient": 0.0,
+            "FFGainCoefficient": 0.0,
+            "FFOffCoefficient": 0.0,
+            "ICoefficient": -0.035,
+            "ILimitMax": 100,
+            "ILimitMin": -50,
+            "Inputs": [
+                "MB_CPU_TEMP_C_4 $bus % 15 * 10"
+            ],
+            "Name": "PID_MB_CPU $bus % 15 * 10",
+            "NegativeHysteresis": 3.0,
+            "OutLimitMax": 100,
+            "OutLimitMin": 0,
+            "PCoefficient": -5.5,
+            "PositiveHysteresis": 0.0,
+            "SetPoint": 74.0,
+            "SlewNeg": 0.0,
+            "SlewPos": 0.0,
+            "Type": "Pid",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "CheckHysteresisWithSetpoint": true,
+            "Class": "temp",
+            "DCoefficient": 0.0,
+            "FFGainCoefficient": 0.0,
+            "FFOffCoefficient": 0.0,
+            "ICoefficient": -0.02,
+            "ILimitMax": 100,
+            "ILimitMin": -50,
+            "Inputs": [
+                "MB_SSD_BOOT_TEMP_C_17 $bus % 15 * 10"
+            ],
+            "Name": "PID_MB_SSD_BOOT $bus % 15 * 10",
+            "NegativeHysteresis": 2.0,
+            "OutLimitMax": 100,
+            "OutLimitMin": 0,
+            "PCoefficient": -1.5,
+            "PositiveHysteresis": 0.0,
+            "SetPoint": 68.0,
+            "SlewNeg": 0.0,
+            "SlewPos": 0.0,
+            "Type": "Pid",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "CheckHysteresisWithSetpoint": true,
+            "Class": "temp",
+            "DCoefficient": 0.0,
+            "FFGainCoefficient": 0.0,
+            "FFOffCoefficient": 0.0,
+            "ICoefficient": -0.02,
+            "ILimitMax": 100,
+            "ILimitMin": -50,
+            "Inputs": [
+                "MB_VR_CPU0_TEMP_C_19 $bus % 15 * 10",
+                "MB_VR_SOC_TEMP_C_20 $bus % 15 * 10",
+                "MB_VR_CPU1_TEMP_C_21 $bus % 15 * 10",
+                "MB_VR_PVDDIO_TEMP_C_22 $bus % 15 * 10",
+                "MB_VR_PVDD11_TEMP_C_23 $bus % 15 * 10"
+            ],
+            "Name": "PID_SD_VR $bus % 15 * 10",
+            "NegativeHysteresis": 3.0,
+            "OutLimitMax": 100,
+            "OutLimitMin": 0,
+            "PCoefficient": -3.0,
+            "PositiveHysteresis": 0.0,
+            "SetPoint": 90.0,
+            "SlewNeg": 0.0,
+            "SlewPos": 0.0,
+            "Type": "Pid",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "Class": "temp",
+            "Inputs": [
+                "MB_INLET_TEMP_C_1 $bus % 15 * 10",
+                "MB_OUTLET_TEMP_C_2 $bus % 15 * 10"
+            ],
+            "Name": "Stepwise_MB_INLET_OUTLET_TEMP $bus % 15 * 10",
+            "NegativeHysteresis": 0,
+            "Output": [
+                20.0
+            ],
+            "PositiveHysteresis": 0,
+            "Reading": [
+                20.0
+            ],
+            "Type": "Stepwise",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "Class": "temp",
+            "Inputs": [
+                "MB_X8_RETIMER_TEMP_C_24 $bus % 15 * 10"
+            ],
+            "Name": "Stepwise_MB_RETIMER_TEMP $bus % 15 * 10",
+            "NegativeHysteresis": 0,
+            "Output": [
+                20.0
+            ],
+            "PositiveHysteresis": 0,
+            "Reading": [
+                20.0
+            ],
+            "Type": "Stepwise",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "Class": "temp",
+            "Inputs": [
+                "Calibrated_MB_FIO_TEMP_C_3 $bus % 15 * 10"
+            ],
+            "Name": "Stepwise_MB_FIO $bus % 15 * 10",
+            "NegativeHysteresis": 2,
+            "Output": [
+                20.0,
+                21.0,
+                22.0,
+                23.0,
+                24.0,
+                25.0,
+                26.0,
+                27.0,
+                28.0,
+                29.0,
+                30.0,
+                31.0,
+                31.0,
+                33.0,
+                35.0,
+                37.0,
+                38.0,
+                39.0,
+                40.0,
+                41.0
+            ],
+            "PositiveHysteresis": 0,
+            "Reading": [
+                20.0,
+                21.0,
+                22.0,
+                23.0,
+                24.0,
+                25.0,
+                26.0,
+                27.0,
+                28.0,
+                29.0,
+                30.0,
+                31.0,
+                32.0,
+                33.0,
+                34.0,
+                35.0,
+                36.0,
+                37.0,
+                38.0,
+                39.0
+            ],
+            "Type": "Stepwise",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "Class": "temp",
+            "Inputs": [
+                "MB_DIMM_A_TEMP_C_5 $bus % 15 * 10",
+                "MB_DIMM_B_TEMP_C_6 $bus % 15 * 10",
+                "MB_DIMM_C_TEMP_C_7 $bus % 15 * 10",
+                "MB_DIMM_D_TEMP_C_8 $bus % 15 * 10",
+                "MB_DIMM_E_TEMP_C_9 $bus % 15 * 10",
+                "MB_DIMM_G_TEMP_C_11 $bus % 15 * 10",
+                "MB_DIMM_H_TEMP_C_12 $bus % 15 * 10",
+                "MB_DIMM_I_TEMP_C_13 $bus % 15 * 10",
+                "MB_DIMM_J_TEMP_C_14 $bus % 15 * 10",
+                "MB_DIMM_K_TEMP_C_15 $bus % 15 * 10"
+            ],
+            "Name": "Stepwise_MB_DIMM $bus % 15 * 10",
+            "NegativeHysteresis": 2,
+            "Output": [
+                20.0,
+                22.0,
+                24.0,
+                26.0,
+                28.0,
+                30.0
+            ],
+            "PositiveHysteresis": 0,
+            "Reading": [
+                65.0,
+                66.0,
+                67.0,
+                68.0,
+                69.0,
+                70.0
+            ],
+            "Type": "Stepwise",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "Class": "temp",
+            "Inputs": [
+                "MB_CPU_TEMP_C_4 $bus % 15 * 10"
+            ],
+            "Name": "Stepwise_MB_CPU $bus % 15 * 10",
+            "NegativeHysteresis": 3,
+            "Output": [
+                20.0,
+                25.0,
+                30.0,
+                35.0,
+                40.0,
+                45.0,
+                50.0,
+                55.0,
+                60.0
+            ],
+            "PositiveHysteresis": 0,
+            "Reading": [
+                66.0,
+                67.0,
+                68.0,
+                69.0,
+                70.0,
+                71.0,
+                72.0,
+                73.0,
+                74.0
+            ],
+            "Type": "Stepwise",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "Class": "temp",
+            "Inputs": [
+                "MB_SSD_BOOT_TEMP_C_17 $bus % 15 * 10"
+            ],
+            "Name": "Stepwise_MB_SSD_BOOT $bus % 15 * 10",
+            "NegativeHysteresis": 2,
+            "Output": [
+                20.0,
+                22.0,
+                24.0,
+                26.0,
+                28.0,
+                30.0
+            ],
+            "PositiveHysteresis": 0,
+            "Reading": [
+                58.0,
+                59.0,
+                60.0,
+                61.0,
+                62.0,
+                63.0
+            ],
+            "Type": "Stepwise",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "Class": "temp",
+            "Inputs": [
+                "MB_VR_CPU0_TEMP_C_19 $bus % 15 * 10",
+                "MB_VR_SOC_TEMP_C_20 $bus % 15 * 10",
+                "MB_VR_CPU1_TEMP_C_21 $bus % 15 * 10",
+                "MB_VR_PVDDIO_TEMP_C_22 $bus % 15 * 10",
+                "MB_VR_PVDD11_TEMP_C_23 $bus % 15 * 10"
+            ],
+            "Name": "Stepwise_SD_VR $bus % 15 * 10",
+            "NegativeHysteresis": 3,
+            "Output": [
+                20.0,
+                22.0,
+                24.0,
+                26.0,
+                28.0,
+                30.0
+            ],
+            "PositiveHysteresis": 0,
+            "Reading": [
+                80.0,
+                81.0,
+                82.0,
+                83.0,
+                84.0,
+                85.0
+            ],
+            "Type": "Stepwise",
+            "Zones": [
+                "Zone 1"
+            ]
+        }
+    ],
+    "Name": "Yosemite 4 Sentinel Dome T1 with Retimer Slot $bus % 15",
+    "Probe": "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': 'Sentinel Dome Retimer.*', 'PRODUCT_PRODUCT_NAME': 'Yosemite V4', 'PRODUCT_INFO_AM2': 'Yosemite V4 T1'})",
+    "Type": "Board",
+    "xyz.openbmc_project.Inventory.Decorator.Asset": {
+        "BuildDate": "$BOARD_MANUFACTURE_DATE",
+        "Manufacturer": "$BOARD_MANUFACTURER",
+        "Model": "$BOARD_PRODUCT_NAME",
+        "PartNumber": "$BOARD_PART_NUMBER",
+        "SerialNumber": "$BOARD_SERIAL_NUMBER",
+        "SparePartNumber": "$BOARD_INFO_AM1"
+    },
+    "xyz.openbmc_project.Inventory.Decorator.AssetTag": {
+        "AssetTag": "$PRODUCT_ASSET_TAG"
+    },
+    "xyz.openbmc_project.Inventory.Decorator.Revision": {
+        "Version": "$PRODUCT_VERSION"
+    },
+    "xyz.openbmc_project.Inventory.Decorator.Slot": {
+        "SlotNumber": "$bus % 15"
+    }
+}
diff --git a/configurations/yosemite4_sentineldome_t2.json b/configurations/yosemite4_sentineldome_t2.json
index 01dba52..15133a2 100644
--- a/configurations/yosemite4_sentineldome_t2.json
+++ b/configurations/yosemite4_sentineldome_t2.json
@@ -9,7 +9,6 @@
             "ICoefficient": -0.035,
             "ILimitMax": 100,
             "ILimitMin": -50,
-            "InputUnavailableAsFailed": false,
             "Inputs": [
                 "MB_DIMM_A_TEMP_C_5 $bus % 15 * 10",
                 "MB_DIMM_B_TEMP_C_6 $bus % 15 * 10",
@@ -24,10 +23,6 @@
                 "MB_DIMM_K_TEMP_C_15 $bus % 15 * 10",
                 "MB_DIMM_L_TEMP_C_16 $bus % 15 * 10"
             ],
-            "MissingIsAcceptable": [
-                "MB_DIMM_F_TEMP_C_10 $bus % 15 * 10",
-                "MB_DIMM_L_TEMP_C_16 $bus % 15 * 10"
-            ],
             "Name": "PID_MB_DIMM $bus % 15 * 10",
             "NegativeHysteresis": 2.0,
             "OutLimitMax": 100,
@@ -103,7 +98,6 @@
             "ICoefficient": -0.02,
             "ILimitMax": 100,
             "ILimitMin": -50,
-            "InputUnavailableAsFailed": false,
             "Inputs": [
                 "MB_SSD_DATA_TEMP_C_18 $bus % 15 * 10"
             ],
@@ -171,29 +165,6 @@
                 "Zone 1"
             ]
         },
-        {
-            "Class": "temp",
-            "InputUnavailableAsFailed": false,
-            "Inputs": [
-                "MB_X8_RETIMER_TEMP_C_24 $bus % 15 * 10"
-            ],
-            "MissingIsAcceptable": [
-                "MB_X8_RETIMER_TEMP_C_24 $bus % 15 * 10"
-            ],
-            "Name": "Stepwise_MB_RETIMER_TEMP $bus % 15 * 10",
-            "NegativeHysteresis": 0,
-            "Output": [
-                20.0
-            ],
-            "PositiveHysteresis": 0,
-            "Reading": [
-                20.0
-            ],
-            "Type": "Stepwise",
-            "Zones": [
-                "Zone 1"
-            ]
-        },
         {
             "Class": "temp",
             "Inputs": [
@@ -253,7 +224,6 @@
         },
         {
             "Class": "temp",
-            "InputUnavailableAsFailed": false,
             "Inputs": [
                 "MB_DIMM_A_TEMP_C_5 $bus % 15 * 10",
                 "MB_DIMM_B_TEMP_C_6 $bus % 15 * 10",
@@ -268,10 +238,6 @@
                 "MB_DIMM_K_TEMP_C_15 $bus % 15 * 10",
                 "MB_DIMM_L_TEMP_C_16 $bus % 15 * 10"
             ],
-            "MissingIsAcceptable": [
-                "MB_DIMM_F_TEMP_C_10 $bus % 15 * 10",
-                "MB_DIMM_L_TEMP_C_16 $bus % 15 * 10"
-            ],
             "Name": "Stepwise_MB_DIMM $bus % 15 * 10",
             "NegativeHysteresis": 2,
             "Output": [
@@ -362,7 +328,6 @@
         },
         {
             "Class": "temp",
-            "InputUnavailableAsFailed": false,
             "Inputs": [
                 "MB_SSD_DATA_TEMP_C_18 $bus % 15 * 10"
             ],
@@ -425,11 +390,7 @@
         }
     ],
     "Name": "Yosemite 4 Sentinel Dome T2 Slot $bus % 15",
-    "Probe": [
-        "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': 'Sentinel Dome', 'PRODUCT_PRODUCT_NAME': 'Yosemite V4', 'PRODUCT_INFO_AM2': 'N/A'})",
-        "OR",
-        "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': 'Sentinel Dome', 'PRODUCT_PRODUCT_NAME': 'Yosemite V4', 'PRODUCT_INFO_AM2': 'Yosemite V4 T2.*'})"
-    ],
+    "Probe": "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': 'Sentinel Dome without Retimer', 'PRODUCT_PRODUCT_NAME': 'Yosemite V4', 'PRODUCT_INFO_AM2': 'Yosemite V4 T2'})",
     "Type": "Board",
     "xyz.openbmc_project.Inventory.Decorator.Asset": {
         "Manufacturer": "$PRODUCT_MANUFACTURER",
diff --git a/configurations/yosemite4_sentineldome_t2_retimer.json b/configurations/yosemite4_sentineldome_t2_retimer.json
new file mode 100644
index 0000000..effe42d
--- /dev/null
+++ b/configurations/yosemite4_sentineldome_t2_retimer.json
@@ -0,0 +1,431 @@
+{
+    "Exposes": [
+        {
+            "CheckHysteresisWithSetpoint": true,
+            "Class": "temp",
+            "DCoefficient": 0.0,
+            "FFGainCoefficient": 0.0,
+            "FFOffCoefficient": 0.0,
+            "ICoefficient": -0.035,
+            "ILimitMax": 100,
+            "ILimitMin": -50,
+            "Inputs": [
+                "MB_DIMM_A_TEMP_C_5 $bus % 15 * 10",
+                "MB_DIMM_B_TEMP_C_6 $bus % 15 * 10",
+                "MB_DIMM_C_TEMP_C_7 $bus % 15 * 10",
+                "MB_DIMM_D_TEMP_C_8 $bus % 15 * 10",
+                "MB_DIMM_E_TEMP_C_9 $bus % 15 * 10",
+                "MB_DIMM_F_TEMP_C_10 $bus % 15 * 10",
+                "MB_DIMM_G_TEMP_C_11 $bus % 15 * 10",
+                "MB_DIMM_H_TEMP_C_12 $bus % 15 * 10",
+                "MB_DIMM_I_TEMP_C_13 $bus % 15 * 10",
+                "MB_DIMM_J_TEMP_C_14 $bus % 15 * 10",
+                "MB_DIMM_K_TEMP_C_15 $bus % 15 * 10",
+                "MB_DIMM_L_TEMP_C_16 $bus % 15 * 10"
+            ],
+            "Name": "PID_MB_DIMM $bus % 15 * 10",
+            "NegativeHysteresis": 2.0,
+            "OutLimitMax": 100,
+            "OutLimitMin": 0,
+            "PCoefficient": -3.0,
+            "PositiveHysteresis": 0.0,
+            "SetPoint": 75.0,
+            "SlewNeg": 0.0,
+            "SlewPos": 0.0,
+            "Type": "Pid",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "CheckHysteresisWithSetpoint": true,
+            "Class": "temp",
+            "DCoefficient": 0.0,
+            "FFGainCoefficient": 0.0,
+            "FFOffCoefficient": 0.0,
+            "ICoefficient": -0.035,
+            "ILimitMax": 100,
+            "ILimitMin": -50,
+            "Inputs": [
+                "MB_CPU_TEMP_C_4 $bus % 15 * 10"
+            ],
+            "Name": "PID_MB_CPU $bus % 15 * 10",
+            "NegativeHysteresis": 3.0,
+            "OutLimitMax": 100,
+            "OutLimitMin": 0,
+            "PCoefficient": -5.5,
+            "PositiveHysteresis": 0.0,
+            "SetPoint": 74.0,
+            "SlewNeg": 0.0,
+            "SlewPos": 0.0,
+            "Type": "Pid",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "CheckHysteresisWithSetpoint": true,
+            "Class": "temp",
+            "DCoefficient": 0.0,
+            "FFGainCoefficient": 0.0,
+            "FFOffCoefficient": 0.0,
+            "ICoefficient": -0.02,
+            "ILimitMax": 100,
+            "ILimitMin": -50,
+            "Inputs": [
+                "MB_SSD_BOOT_TEMP_C_17 $bus % 15 * 10"
+            ],
+            "Name": "PID_MB_SSD_BOOT $bus % 15 * 10",
+            "NegativeHysteresis": 2.0,
+            "OutLimitMax": 100,
+            "OutLimitMin": 0,
+            "PCoefficient": -1.5,
+            "PositiveHysteresis": 0.0,
+            "SetPoint": 68.0,
+            "SlewNeg": 0.0,
+            "SlewPos": 0.0,
+            "Type": "Pid",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "CheckHysteresisWithSetpoint": true,
+            "Class": "temp",
+            "DCoefficient": 0.0,
+            "FFGainCoefficient": 0.0,
+            "FFOffCoefficient": 0.0,
+            "ICoefficient": -0.02,
+            "ILimitMax": 100,
+            "ILimitMin": -50,
+            "Inputs": [
+                "MB_SSD_DATA_TEMP_C_18 $bus % 15 * 10"
+            ],
+            "Name": "PID_MB_SSD_DATA $bus % 15 * 10",
+            "NegativeHysteresis": 2.0,
+            "OutLimitMax": 100,
+            "OutLimitMin": 0,
+            "PCoefficient": -1.5,
+            "PositiveHysteresis": 0.0,
+            "SetPoint": 68.0,
+            "SlewNeg": 0.0,
+            "SlewPos": 0.0,
+            "Type": "Pid",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "CheckHysteresisWithSetpoint": true,
+            "Class": "temp",
+            "DCoefficient": 0.0,
+            "FFGainCoefficient": 0.0,
+            "FFOffCoefficient": 0.0,
+            "ICoefficient": -0.02,
+            "ILimitMax": 100,
+            "ILimitMin": -50,
+            "Inputs": [
+                "MB_VR_CPU0_TEMP_C_19 $bus % 15 * 10",
+                "MB_VR_SOC_TEMP_C_20 $bus % 15 * 10",
+                "MB_VR_CPU1_TEMP_C_21 $bus % 15 * 10",
+                "MB_VR_PVDDIO_TEMP_C_22 $bus % 15 * 10",
+                "MB_VR_PVDD11_TEMP_C_23 $bus % 15 * 10"
+            ],
+            "Name": "PID_SD_VR $bus % 15 * 10",
+            "NegativeHysteresis": 3.0,
+            "OutLimitMax": 100,
+            "OutLimitMin": 0,
+            "PCoefficient": -3.0,
+            "PositiveHysteresis": 0.0,
+            "SetPoint": 90.0,
+            "SlewNeg": 0.0,
+            "SlewPos": 0.0,
+            "Type": "Pid",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "Class": "temp",
+            "Inputs": [
+                "MB_INLET_TEMP_C_1 $bus % 15 * 10",
+                "MB_OUTLET_TEMP_C_2 $bus % 15 * 10"
+            ],
+            "Name": "Stepwise_MB_INLET_OUTLET_TEMP $bus % 15 * 10",
+            "NegativeHysteresis": 0,
+            "Output": [
+                20.0
+            ],
+            "PositiveHysteresis": 0,
+            "Reading": [
+                20.0
+            ],
+            "Type": "Stepwise",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "Class": "temp",
+            "Inputs": [
+                "MB_X8_RETIMER_TEMP_C_24 $bus % 15 * 10"
+            ],
+            "Name": "Stepwise_MB_RETIMER_TEMP $bus % 15 * 10",
+            "NegativeHysteresis": 0,
+            "Output": [
+                20.0
+            ],
+            "PositiveHysteresis": 0,
+            "Reading": [
+                20.0
+            ],
+            "Type": "Stepwise",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "Class": "temp",
+            "Inputs": [
+                "Calibrated_MB_FIO_TEMP_C_3 $bus % 15 * 10"
+            ],
+            "Name": "Stepwise_MB_FIO $bus % 15 * 10",
+            "NegativeHysteresis": 2,
+            "Output": [
+                20.0,
+                21.0,
+                22.0,
+                23.0,
+                24.0,
+                25.0,
+                26.0,
+                27.0,
+                28.0,
+                29.0,
+                30.0,
+                31.0,
+                31.0,
+                33.0,
+                35.0,
+                37.0,
+                38.0,
+                39.0,
+                40.0,
+                41.0
+            ],
+            "PositiveHysteresis": 0,
+            "Reading": [
+                20.0,
+                21.0,
+                22.0,
+                23.0,
+                24.0,
+                25.0,
+                26.0,
+                27.0,
+                28.0,
+                29.0,
+                30.0,
+                31.0,
+                32.0,
+                33.0,
+                34.0,
+                35.0,
+                36.0,
+                37.0,
+                38.0,
+                39.0
+            ],
+            "Type": "Stepwise",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "Class": "temp",
+            "Inputs": [
+                "MB_DIMM_A_TEMP_C_5 $bus % 15 * 10",
+                "MB_DIMM_B_TEMP_C_6 $bus % 15 * 10",
+                "MB_DIMM_C_TEMP_C_7 $bus % 15 * 10",
+                "MB_DIMM_D_TEMP_C_8 $bus % 15 * 10",
+                "MB_DIMM_E_TEMP_C_9 $bus % 15 * 10",
+                "MB_DIMM_F_TEMP_C_10 $bus % 15 * 10",
+                "MB_DIMM_G_TEMP_C_11 $bus % 15 * 10",
+                "MB_DIMM_H_TEMP_C_12 $bus % 15 * 10",
+                "MB_DIMM_I_TEMP_C_13 $bus % 15 * 10",
+                "MB_DIMM_J_TEMP_C_14 $bus % 15 * 10",
+                "MB_DIMM_K_TEMP_C_15 $bus % 15 * 10",
+                "MB_DIMM_L_TEMP_C_16 $bus % 15 * 10"
+            ],
+            "Name": "Stepwise_MB_DIMM $bus % 15 * 10",
+            "NegativeHysteresis": 2,
+            "Output": [
+                20.0,
+                22.0,
+                24.0,
+                26.0,
+                28.0,
+                30.0
+            ],
+            "PositiveHysteresis": 0,
+            "Reading": [
+                65.0,
+                66.0,
+                67.0,
+                68.0,
+                69.0,
+                70.0
+            ],
+            "Type": "Stepwise",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "Class": "temp",
+            "Inputs": [
+                "MB_CPU_TEMP_C_4 $bus % 15 * 10"
+            ],
+            "Name": "Stepwise_MB_CPU $bus % 15 * 10",
+            "NegativeHysteresis": 3,
+            "Output": [
+                20.0,
+                25.0,
+                30.0,
+                35.0,
+                40.0,
+                45.0,
+                50.0,
+                55.0,
+                60.0
+            ],
+            "PositiveHysteresis": 0,
+            "Reading": [
+                66.0,
+                67.0,
+                68.0,
+                69.0,
+                70.0,
+                71.0,
+                72.0,
+                73.0,
+                74.0
+            ],
+            "Type": "Stepwise",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "Class": "temp",
+            "Inputs": [
+                "MB_SSD_BOOT_TEMP_C_17 $bus % 15 * 10"
+            ],
+            "Name": "Stepwise_MB_SSD_BOOT $bus % 15 * 10",
+            "NegativeHysteresis": 2,
+            "Output": [
+                20.0,
+                22.0,
+                24.0,
+                26.0,
+                28.0,
+                30.0
+            ],
+            "PositiveHysteresis": 0,
+            "Reading": [
+                58.0,
+                59.0,
+                60.0,
+                61.0,
+                62.0,
+                63.0
+            ],
+            "Type": "Stepwise",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "Class": "temp",
+            "Inputs": [
+                "MB_SSD_DATA_TEMP_C_18 $bus % 15 * 10"
+            ],
+            "Name": "Stepwise_MB_SSD_DATA $bus % 15 * 10",
+            "NegativeHysteresis": 2,
+            "Output": [
+                20.0,
+                22.0,
+                24.0,
+                26.0,
+                28.0,
+                30.0
+            ],
+            "PositiveHysteresis": 0,
+            "Reading": [
+                58.0,
+                59.0,
+                60.0,
+                61.0,
+                62.0,
+                63.0
+            ],
+            "Type": "Stepwise",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "Class": "temp",
+            "Inputs": [
+                "MB_VR_CPU0_TEMP_C_19 $bus % 15 * 10",
+                "MB_VR_SOC_TEMP_C_20 $bus % 15 * 10",
+                "MB_VR_CPU1_TEMP_C_21 $bus % 15 * 10",
+                "MB_VR_PVDDIO_TEMP_C_22 $bus % 15 * 10",
+                "MB_VR_PVDD11_TEMP_C_23 $bus % 15 * 10"
+            ],
+            "Name": "Stepwise_SD_VR $bus % 15 * 10",
+            "NegativeHysteresis": 3,
+            "Output": [
+                20.0,
+                22.0,
+                24.0,
+                26.0,
+                28.0,
+                30.0
+            ],
+            "PositiveHysteresis": 0,
+            "Reading": [
+                80.0,
+                81.0,
+                82.0,
+                83.0,
+                84.0,
+                85.0
+            ],
+            "Type": "Stepwise",
+            "Zones": [
+                "Zone 1"
+            ]
+        }
+    ],
+    "Name": "Yosemite 4 Sentinel Dome T2 with Retimer Slot $bus % 15",
+    "Probe": "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': 'Sentinel Dome Retimer.*', 'PRODUCT_PRODUCT_NAME': 'Yosemite V4', 'PRODUCT_INFO_AM2': 'Yosemite V4 T2'})",
+    "Type": "Board",
+    "xyz.openbmc_project.Inventory.Decorator.Asset": {
+        "BuildDate": "$BOARD_MANUFACTURE_DATE",
+        "Manufacturer": "$BOARD_MANUFACTURER",
+        "Model": "$BOARD_PRODUCT_NAME",
+        "PartNumber": "$BOARD_PART_NUMBER",
+        "SerialNumber": "$BOARD_SERIAL_NUMBER",
+        "SparePartNumber": "$BOARD_INFO_AM1"
+    },
+    "xyz.openbmc_project.Inventory.Decorator.AssetTag": {
+        "AssetTag": "$PRODUCT_ASSET_TAG"
+    },
+    "xyz.openbmc_project.Inventory.Decorator.Revision": {
+        "Version": "$PRODUCT_VERSION"
+    },
+    "xyz.openbmc_project.Inventory.Decorator.Slot": {
+        "SlotNumber": "$bus % 15"
+    }
+}
diff --git a/meson.build b/meson.build
index 8bf7966..1baa872 100644
--- a/meson.build
+++ b/meson.build
@@ -198,7 +198,9 @@ configs = [
     'yosemite4_fanboard_fsc_max.json',
     'yosemite4_fanboard_fsc_nct.json',
     'yosemite4_sentineldome_t1.json',
+    'yosemite4_sentineldome_t1_retimer.json',
     'yosemite4_sentineldome_t2.json',
+    'yosemite4_sentineldome_t2_retimer.json',
     'yosemite4_floatingfalls.json',
     'yosemite4_wailuafalls.json',
     'yosemite4.json',
-- 
2.25.1

